### ISSUE TITLE
Insomnia reporting: oas3-valid-oas-content-example 'example' property should match exactly one schema in oneOf 

### ISSUE THREAD

**Issue by Weshnaw** [getinsomnia/insomnia#4683]
### Expected Behavior

No warning message

### Actual Behavior

oas3-valid-oas-content-example 'example' property should match exactly one schema in oneOf warning appears

### Reproduction Steps

1. use the following spec in Insomnia's document design tab

<details>

<summary>example OAS</summary>

```json
{
  "openapi": "3.0.3",
  "info": {
    "title": "playground-api",
    "version": "0.1.0"
  },
  "paths": {
    "/todo": {
      "post": {
        "tags": [
          "todo"
        ],
        "summary": "Create new Todo to shared in-memory storage.",
        "description": "Create new Todo",
        "operationId": "create_todo",
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/Todo"
              }
            }
          },
          "required": true
        },
        "responses": {
          "201": {
            "description": "Todo created successfully",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Todo"
                }
              }
            }
          },
          "409": {
            "description": "Todo with id already exists",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/TodoErrorResponse"
                },
                "example": {
                  "Conflict": "id = 1"
                }
              }
            }
          }
        },
        "deprecated": false
      }
    }
  },
  "components": {
    "schemas": {
      "TodoErrorResponse": {
        "oneOf": [
          {
            "type": "object",
            "properties": {
              "NotFound": {
                "type": "string",
                "description": "When Todo is not found by search term."
              }
            }
          },
          {
            "type": "object",
            "properties": {
              "Conflict": {
                "type": "string",
                "description": "When there is a conflict storing a new todo."
              }
            }
          },
          {
            "type": "object",
            "properties": {
              "Unauhtorized": {
                "type": "string",
                "description": "When todo enpoint was called without correct credentials"
              }
            }
          }
        ],
        "description": "Todo endpoint error responses"
      },
      "Todo": {
        "type": "object",
        "required": [
          "id",
          "value",
          "checked"
        ],
        "properties": {
          "id": {
            "type": "integer",
            "format": "int32",
            "description": "Unique id for the todo item.",
            "example": 1
          },
          "checked": {
            "type": "boolean",
            "description": "Mark is the task done or not"
          },
          "value": {
            "type": "string",
            "description": "Description of the taks to do.",
            "example": "Remember to buy groceries"
          }
        },
        "description": "Task to do."
      }
    }
  },
  "tags": [
    {
      "name": "todo",
      "description": "Todo management endpoints."
    }
  ]
}
```

</details>

2. View warning related to line 43

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

I am no expert at OpenAPI specs, however, from my limited understanding I see the example at line 43 is valid, if not please let me know what the correct formatting is.

### Insomnia Version

2022.2.1

### What operating system are you using?

Windows

### Operating System Version

Windows 11 Pro 21H2

### Installation method

Downloaded from insomnia.rest

**Comment by dimitropoulos** [comment]
We took a good look at this today on [the Insomnia stream](https://youtu.be/n5nsA_fJ4vI?t=3009).  I'm stumped.  I will say, this is something we did notice is that:
1. the property is named `examples` (not `example`) from what I can tell
1. inside the example, you provide a few properties ([see the spec](https://swagger.io/specification/#:~:text=callback%20successfully%20processed-,Example%20Object,-Fixed%20Fields)) _one_ of which is `value`.
1. you need to give the example a name (e.g. `foo` on the OAS docs lined in the above)
1. Insomnia _does_ seem to be seeing the schema, but something's not right...

the best I could come up with was:
```json
          "409": {
            "description": "Todo with id already exists",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/TodoErrorResponse"
                },
                "examples": {
                  "foo": {
                    "value": {
                      "NotFound": "id = 1"
                    }
                  }
                }
              }
            }
          }
``` 
          
which appears to me to be what the OAS spec calls for.. Any thoughts?

**Comment by Weshnaw** [comment]
The stream clip was very informative (I wasn't aware of the details tag).

There's just a couple things I could add:

While originally searching this issue I did notice that quite a few 'example' examples showed using the plural 'examples' in the way that you describe, but I also found that singular 'example' should still be valid ([so says swagger](https://swagger.io/docs/specification/adding-examples/)).

As you mentioned in your stream this seems to be entirely just a linting error, and it doesn't really effect anything other then just having to see a warning, as the swagger ui correctly generates and shows everything as expected.

I also believe that the issue is specifically related to using examples with a schema that utilizes oneOf (Though I could be wrong on this, and I haven't really tested if it effects any of the other enum style schema's,)
